main.py

```diff
@@ -303,6 +303,14 @@ def index(marker=None, message=None):
             context['map_only'] = 1
     if 'lat' in request.values and 'lon' in request.values:
         context['coordinates'] = (request.values['lat'], request.values['lon'])
+    for attr in 'approx', 'accurate', 'show_markers', 'show_discussions', 'show_urban', 'show_intersection', 'show_lane',\
+                'show_day', 'show_holiday', 'show_time', 'start_time', 'end_time', 'weather', 'road', 'separation',\
+                'surface', 'acctype', 'controlmeasure', 'district', 'case_type', 'show_fatal', 'show_severe', 'show_light':
+        if attr in request.values:
```

Comment:
Instead of querying the dictionary three times, I think it's better to do:

    value = request.values.get(attr)
    if value is not None:
        context[attr] = value or -1

What do you think?
In case the value is `0`, should we take the value itself or `-1`?
Because generally saying `not value` is more pythonic than `value == ''`, but it is also true when `value == 0`.